src/runtime/schema.ts

```diff
@@ -30,7 +30,7 @@ export class Schema {
   // in `fields`; the constructor will convert these to the correct schema form.
   // tslint:disable-next-line: no-any
   constructor(names: string[], fields: Dictionary<any>, description?) {
-    this.names = names;
+    this.names = names.filter((value => !!value));
```

Comment:
You have fixed the symptom but not the underlying issue. Why was the list of names `[undefined]` in the first place?

It is true that passing a an array with undefined, null or an empty string is wrong here, but we cannot do full validation in every constructor and in every method - that would be too costly.

It is best to fix the problem at its root, and the actual bug is in processing of the parsed output in `manifest.ts`. The `_processSchema` method has such a funky line in it:

`names = [names[0], ...names.filter(name => name !== names[0])];`

This looks to me like a source of our pain. A name array can be empty, but after this line it will be a one element array of undefined.

What this line attempts to achieve (I think) is avoiding duplicate names I presume, I would replace it with:

`names = [...new Set(names)];`

Reply:
According to some of my other tests, this does not cause the problem. I traced the issue I was experiencing to code I introduced in another PR. 

This suggestion is a valid improvement.